container/check_container.go

```diff
@@ -55,44 +86,30 @@ func (c *containerCheck) Run(ctx context.Context) (certification.Results, error)
 
 		override, err := lib.GetContainerPolicyExceptions(ctx, p)
 		if err != nil {
-			return certification.Results{}, fmt.Errorf("%w: %s", preflighterr.ErrCannotResolvePolicyException, err)
+			return fmt.Errorf("%w: %s", preflighterr.ErrCannotResolvePolicyException, err)
 		}
 
-		pol = override
+		c.policy = override
 	}
-
-	checks, err := engine.InitializeContainerChecks(ctx, pol, engine.ContainerCheckConfig{
+	var err error
+	c.checks, err = engine.InitializeContainerChecks(ctx, c.policy, engine.ContainerCheckConfig{
 		DockerConfig:           c.dockerconfigjson,
 		PyxisAPIToken:          c.pyxisToken,
 		CertificationProjectID: c.certificationProjectID,
 	})
 	if err != nil {
-		return certification.Results{}, fmt.Errorf("%w: %s", preflighterr.ErrCannotInitializeChecks, err)
+		return fmt.Errorf("%w: %s", preflighterr.ErrCannotInitializeChecks, err)
 	}
+	c.resolved = true
 
-	cfg := runtime.Config{
-		Image:              c.image,
-		DockerConfig:       c.dockerconfigjson,
-		Scratch:            pol == policy.PolicyScratch,
-		Bundle:             false,
-		Insecure:           c.insecure,
-		Platform:           c.platform,
-		ManifestListDigest: c.manifestListDigest,
-	}
-	eng, err := engine.New(ctx, checks, nil, cfg)
-	if err != nil {
-		return certification.Results{}, err
-	}
-
-	if err != nil {
-		return certification.Results{}, err
-	}
-
-	if err := eng.ExecuteChecks(ctx); err != nil {
-		return certification.Results{}, err
-	}
+	return nil
+}
 
-	return eng.Results(ctx), nil
+// List the available container checks. Policy exceptions will be resolved if the proper
+// pyxis information is provided.
+func (c *containerCheck) List(ctx context.Context) (policy.Policy, []check.Check, error) {
+	err := c.resolve(ctx)
+	return c.policy, c.checks, err
```

Comment:
If you aren't going to check the error here, you might as well just inline it all into the return statement.

Reply:
fixed